cmd/reference-addon-manager/main.go

```diff
@@ -92,11 +93,40 @@ func main() {
 		}
 	}
 
+	// fetching the following from the Downwards Api in case of reference-addon.
```

Comment:
Nit, but you can only set the `ADDON_NAMESPACE` environment variable, using the kubernetes downwards API.
I would not mention the downwards api at all, because we first have to settle on how to provide these variables.
In best case, the Addon Operator can just set these via the Subscription and we don't need Addon authors to worry about these parameters at all.
Also the `/config/deploy/deployment.tpl.yaml` is currently missing both of these environment variables.